tensorflow_probability/python/distributions/sratio.py

```diff
@@ -0,0 +1,359 @@
+      dtype: The type of the event samples (default: int32).
+      validate_args: Python `bool`, default `False`. When `True` distribution
+        parameters are checked for validity despite possibly degrading runtime
+        performance. When `False` invalid inputs may silently render incorrect
+        outputs.
+      allow_nan_stats: Python `bool`, default `True`. When `True`, statistics
+        (e.g. mode) use the value "`NaN`" to indicate the result is
+        undefined. When `False`, an exception is raised if one or more of the
+        statistic's batch members are undefined.
+      name: Python `str` name prefixed to Ops created by this class.
+    """
+
+    parameters = dict(locals())
+
+    with tf.name_scope(name) as name:
+
+      float_dtype = dtype_util.common_dtype(
+          [cutpoints, loc],
+          dtype_hint=tf.float32)
+
+      self._cutpoints = tensor_util.convert_nonref_to_tensor(
+          cutpoints, dtype_hint=float_dtype, name='cutpoints')
+      self._loc = tensor_util.convert_nonref_to_tensor(
+          loc, dtype_hint=float_dtype, name='loc')
+
+      super(StoppingRatioLogistic, self).__init__(
+          dtype=dtype,
+          reparameterization_type=reparameterization.NOT_REPARAMETERIZED,
+          validate_args=validate_args,
+          allow_nan_stats=allow_nan_stats,
+          parameters=parameters,
+          name=name)
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(cutpoints=1, loc=0)
+
+  @staticmethod
+  def _param_shapes(sample_shape):
+    return dict(
+        zip(('loc', 'scale'),
+            ([tf.convert_to_tensor(sample_shape, dtype=tf.int32)] * 2)))
+
+  @property
+  def cutpoints(self):
+    """Input argument `cutpoints`."""
+    return self._cutpoints
+
+  @property
+  def loc(self):
+    """Input argument `loc`."""
+    return self._loc
+
+  def categorical_log_probs(self):
+    """Log probabilities for the `K+1` sequential categories."""
+    return tf.math.log(self.categorical_probs())
+
+  def categorical_probs(self):
+    """Probabilities for the `K+1` sequential categories."""
+    cutpoints = self.cutpoints
+    loc = self.loc
+
+    num_cat = self._num_categories()
+    one = tf.ones(1, dtype=cutpoints.dtype)
+
+    q = one - tf.math.sigmoid(cutpoints - loc[..., tf.newaxis])
+    p = one - q
+
+    qs = tf.math.cumprod(q[..., :(num_cat - 2)], axis=-1)
+    p = tf.concat([p[..., :1], p[..., 1:num_cat] * qs], axis=-1)
+    qs = tf.math.reduce_prod(q[..., :num_cat], axis=-1)
+    p = tf.concat([p, qs[..., tf.newaxis]], axis=-1)
+
+    return p
+
+  def _num_categories(self):
+    return prefer_static.shape(self.cutpoints, out_type=self.dtype)[-1] + 1
+
+  def _sample_n(self, n, seed=None):
+    logits = tf.reshape(
+        self.categorical_log_probs(), [-1, self._num_categories()])
+    draws = tf.random.categorical(logits, n, dtype=self.dtype, seed=seed)
+    return tf.reshape(
+        tf.transpose(draws),
```

Comment:
Can we avoid this transpose? Generally we try to avoid `tf.transpose` as much as possible for performance reasons.

Reply:
I don't think so. I adopted it from the `Categorical` distribution (see [here](https://github.com/tensorflow/probability/blob/master/tensorflow_probability/python/distributions/categorical.py#L244)) which does the same.